politeiawww/proposals.go

```diff
@@ -890,6 +899,89 @@ func (p *politeiawww) processProposalDetails(propDetails www.ProposalsDetails, u
 	return &reply, nil
 }
 
+// voteSummaries fetches the voting summary information for a set of
+// proposals.
+func (p *politeiawww) voteSummaries(tokens []string, bestBlock uint64) (map[string]www.VoteSummary, error) {
+	r, err := p.decredBatchVoteSummary(tokens)
+	if err != nil {
+		return nil, err
+	}
+
+	voteSummaries := make(map[string]www.VoteSummary)
+
+	for token, summary := range r.Summaries {
+		results := convertVoteOptionResultsFromDecred(summary.Results)
+
+		endHeight, err := strconv.ParseUint(summary.EndHeight, 10, 64)
+		if err != nil {
+			endHeight = 0
+		}
+
+		vs := www.VoteSummary{
+			Status:           voteStatusFromVoteSummary(summary, bestBlock),
+			EligibleTickets:  uint32(summary.EligibleTicketCount),
+			EndHeight:        endHeight,
+			QuorumPercentage: summary.QuorumPercentage,
+			PassPercentage:   summary.PassPercentage,
+			Results:          results,
+		}
+
+		voteSummaries[token] = vs
+	}
+
+	return voteSummaries, nil
+}
+
+// processBatchVoteSummary the voting summary information for a set of
+// proposals.
+func (p *politeiawww) processBatchVoteSummary(batchVoteSummary www.BatchVoteSummary) (*www.BatchVoteSummaryReply, error) {
```

Comment:
If you look at the `processVoteStatus` function you'll notice that we memoize the vote results for proposals that have finished voting in a memory cache in order to improve performance.  We need to do the same thing here.

The problem you're going to run into is that the `VoteResult` and `VoteStatusReply` don't map 1-to-1.  The way I would do this would be to convert the `p.voteStatusReply` memory cache to hold vote results instead then update the vote status endpoint to cache a `VoteResult` instead of a `VoteStatusReply`.  This way, when we go to remove the deprecated vote status endpoints we don't have to worry about updating the memory cache.  We can just delete the endpoints.

Reply:
I don't see any `VoteResult` type anywhere. I can cache `decredplugin.VoteSummaryReply` though.

Reply:
That's kind of strange too though... i think it would be the cleanest to just cache both the`VoteSummary` and the `VoteStatusReply` and then remove the `VoteStatusReply` cache when deprecating that route, no?

Reply:
> I don't see any VoteResult type anywhere

Typo.  I meant `VoteSummary`.

> i think it would be the cleanest to just cache both theVoteSummary and the VoteStatusReply and then remove the VoteStatusReply cache when deprecating that route, no?

I'm really not a fan of having two caches with the same data in them.  I think its cleaner to have just a `VoteSummary` cache, since the vote status routes are being deprecated, and then to have helper functions like the ones below for vote status routes until they're fully removed.

```
func (p *politeiawww) cacheVoteStatusReply(v www.VoteStatusReply) {
	// Convert VoteStatusReply to VoteSummary
	// Cache VoteSummary
}

func (p *politeiawww) getVoteStatusReply(token string) (*www.VoteStatusReply, bool) {
	// Lookup VoteSummary for token
	// Convert VoteSummary to VoteStatusReply
}
```